CMakeLists.txt

```diff
@@ -263,145 +264,159 @@ endif ()
 
 set (FFTW_VERSION 3.3.7)
 
-set (PREC_SUFFIX)
-if (ENABLE_FLOAT)
-  set (FFTW_SINGLE TRUE)
-  set (BENCHFFT_SINGLE TRUE)
-  set (PREC_SUFFIX f)
+if (BUILD_SHARED_LIBS)
+  add_definitions (-DFFTW_DLL)
 endif ()
 
-if (ENABLE_LONG_DOUBLE)
-  set (FFTW_LDOUBLE TRUE)
-  set (BENCHFFT_LDOUBLE TRUE)
-  set (PREC_SUFFIX l)
-endif ()
+function (add_fftw_library PRECISION)
+  if (PRECISION MATCHES "SINGLE|LDOUBLE|QUAD")
+    set (FFTW_${PRECISION} TRUE)
+    set (BENCHFFT_${PRECISION} TRUE)
+    if (PRECISION STREQUAL SINGLE)
```

Comment:
You should update requirement to CMake 3.1+ or use `string(COMPARE EQUAL ...)`:
* https://cgold.readthedocs.io/en/latest/tutorials/control-structures.html#cmp0054

Reply:
This problem looks like it will affect the `if (PRECISION MATCHES "...")` line too, so I guess it's better to update to 3.1.